tools/about_paint.py

```diff
@@ -0,0 +1,117 @@
+from collections import Counter
+import copy
+from fontTools.colorLib.builder import LayerListBuilder
+from fontTools.ttLib import TTFont
+from fontTools.ttLib.tables import otTables as ot
+from collections import defaultdict
+from itertools import groupby
+import logging
+from math import acos, asin, atan, degrees
+import os
+import sys
+
+def traverse_paint(colr, callback):
+    """Traverse and optionally rewrite Paint in a COLRv1 graph."""
+    if not callable(callback):
+        raise TypeError("callback must be callable")
+
+    visited = set()
+    frontier = []
+    # Glyph graphs start from either base glyphs or the layerlist
```

Comment:
glyph graphs _only_ start from base glyphs, not from layer list. If any Paint in LayerList is not referenced by any base glyph (unlikely), it's as if it's not even there

Reply:
It would be valid but pointless to have a stranded glyph in the layer list I think?

Reply:
indeed, but my point was simply that one normally should start traversing the COLR layer graphs from the root of a BaseGlyph record which is the only entry point for a renderer.

Reply:
you already keep tally of the visited paints. for a normal well-behaved font (containing no dead paints in the LayerList) the end result of the traversal should be the same whether you include or not the top-level paints in the LayerList. I lean towards only including the root paints in the base glyph records.

Reply:
I think here - unlike in a renderer - I do care about unreachable Paint since I just want to know what all is in the table?

Reply:
>  I just want to know what all is in the table

good point